python/eight_mile/tf/layers.py

```diff
@@ -1347,27 +1324,40 @@ def neg_log_loss(self, unary, tags, lengths):
 class LangSequenceModel(tf.keras.Model):
 
     def __init__(self, nc, embeddings, transducer, decoder=None, name=None):
-        super(LangSequenceModel, self).__init__(name=name)
+        super().__init__(name=name)
         if isinstance(embeddings, dict):
             self.embed_model = EmbeddingsStack(embeddings)
         else:
             assert isinstance(embeddings, EmbeddingsStack)
             self.embed_model = embeddings
         self.transducer_model = transducer
-
+        if hasattr(transducer, 'requires_state') and transducer.requires_state:
```

Comment:
`if getattr(transducer, 'requires_state', False):` also works, is this more explicit?